bowtie/_cli.py

```diff
@@ -986,6 +992,11 @@ async def filter_implementations(
     """
     Output implementations matching a given criteria.
     """
+    if languages == KNOWN_LANGUAGES:
```

Comment:
I haven't tried to run this but it looks suspicious.

Doesn't it ignore the other filter? I.e. if you provide a dialect filter doesn't this now output the wrong answer as it will output implementations regardless of whether they support the given dialects.

Reply:
Yeah I did think of this and was going to do it as `if not dialects and languages == KNOWN_LANGUAGES` but the `not dialects` check here was redundant since I've made it sure in the `callback` that this won't happen because the callback straight away returns KNOWN_LANGUAGES and doesn't really perform the below logic:

```python
  LANGUAGE_ALIASES.get(each, each)  # type: ignore[reportUnknownArgumentType]
  for each in value or KNOWN_LANGUAGES
```

But still to make the code more readable I can add the dialect check as well i.e. `if not dialects and languages == KNOWN_LANGUAGES`

Reply:
> Doesn't it ignore the other filter? I.e. if you provide a dialect filter doesn't this now output the wrong answer as it will output implementations regardless of whether they support the given dialects.

Also the tests would've failed in that case

Reply:
```python
        KNOWN_LANGUAGES
        if not value
        and not ctx.params["dialects"]
```

in here in the `callback` the `not dialects` check happens.

Reply:
Got it -- I think it'd be slightly clearer for this check to happen in one place? Do you see an easy way to do that bit here too?

It'd make sense to do it in the callback if we didn't have to do any `if` here but you're doing one anyways to check against `KNOWN_LANGUAGES`.